app/js/services/QueryBuilderService.js

```diff
@@ -1,95 +1,118 @@
-function QueryBuilderService($resource, $http, AppSettings, NotificationService, _) {
-  'ngInject';
-
-  const QUERY_ENDPOINT = `${AppSettings.api.baseUrl}${AppSettings.api.classifiers}/`;
-  let USER_ENDPOINT = (slug) => `${AppSettings.api.baseUrl}${AppSettings.api.users}/${slug}/`;
-  const IGNORE_EXISTING_EMAIL = true
+function QueryBuilderService(
+  $resource,
+  $http,
+  AppSettings,
+  NotificationService,
+  _
+) {
+  "ngInject";
+
+  const QUERY_ENDPOINT = `${AppSettings.api.baseUrl}${
+    AppSettings.api.classifiers
+  }/`;
+  let USER_ENDPOINT = slug =>
+    `${AppSettings.api.baseUrl}${AppSettings.api.users}/${slug}/`;
+  const IGNORE_EXISTING_EMAIL = true;
 
   const service = {
-    submitQuery: function(diseases = [], genes =[], user){
+    submitQuery: function(diseases = [], genes = [], user) {
       if (user === null || user === undefined) {
         NotificationService.notify({
-          type:    'error',
+          type: "error",
           message: `User is null or undefined.`
         });
-        return
-      } else if (user.random_slugs === null || user.random_slugs === undefined) {
+        return;
+      } else if (
+        user.random_slugs === null ||
+        user.random_slugs === undefined
+      ) {
         NotificationService.notify({
-          type:    'error',
+          type: "error",
           message: `random_slugs is null or undefined.`
         });
-        return
+        return;
       } else if (user.random_slugs.length < 1) {
         NotificationService.notify({
-          type:    'error',
+          type: "error",
           message: `random_slugs is empty.`
         });
-        return
+        return;
       }
 
-      let queryResource = $resource(QUERY_ENDPOINT, {}, {
-                submit: {
-                  method: 'POST',
-                  headers: {
-                    'Authorization': `Bearer ${user.random_slugs[0]}`,
-                  }
-                }
-              });
+      let queryResource = $resource(
+        QUERY_ENDPOINT,
+        {},
+        {
+          submit: {
+            method: "POST",
+            headers: {
+              Authorization: `Bearer ${user.random_slugs[0]}`
+            }
+          }
+        }
+      );
 
       // Helper function to get the total number of positive or negative samples in a query
-      let getTotalsFor = setParam=>_.reduce(_.pluck(diseases, setParam), (a,b)=>a+b)
+      let getTotalsFor = setParam =>
+        _.reduce(_.pluck(diseases, setParam), (a, b) => a + b);
 
       function validateClassifier() {
-        let numberPositives = getTotalsFor('positives')
-        let numberNegatives = getTotalsFor('negatives')
+        let numberPositives = getTotalsFor("positives");
+        let numberNegatives = getTotalsFor("negatives");
         // 20 positive samples and 20 negative samples
         if (numberPositives >= 20 && numberNegatives >= 20) {
-          return true
+          return true;
```

Comment:
Seems that this `if` block can be removed because it does nothing.

Reply:
I think we still need this because the function should return true if there are 20 positive sample and 20 negative samples. If not, the notification fires.